var/spack/repos/builtin/packages/perl-bioperl/package.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PerlBioperl(PerlPackage):
```

Comment:
I see a `perl-bio-perl` package. Is this related?

Reply:
Hmm, I think that may be an older version, where 1.007002 == 1.7.2. I would have to look into this more. It could be that the host repository has changed, along with the version numbering scheme. Even still, based on the tarfile name, I am not sure why there is a '-' in 'bio-perl'.

Reply:
The naming is frequently based off of the camel case in the URL. In this case the `https://cpan.metacpan.org/authors/id/C/CD/CDRAUG/BioPerl-1.7.6.tar.gz` contains `BioPerl-1.7.6.tar.gz` and if you name this class accordingly (`PerlBioPerl`) this is referred to on the command line as `perl-bio-perl` (a dash is inserted between "words" where the capital letter separates words).

A similar application would lead to naming the Termreadkey package `PerlTermReadKey` and it would be referred to as `perl-term-read-key` but I think whoever created it didn't want the extra dash. The consistent approach would be to name the package based on the homepage project name.

There is some leeway in how packages are named (other than that e.g. Perl packages should start with `Perl...`) so these minor inconsistencies don't bother me, as long as we end up with one Spack package per actual package.

Reply:
Looking at the `README` and `CHANGES` files it does appear that 1.007002==1.7.2. Those files refer to the version as 1.7.2. So, I will combine my new file into the existing one.

Reply:
>A similar application would lead to naming the Termreadkey package PerlTermReadKey and it would be referred to as perl-term-read-key but I think whoever created it didn't want the extra dash.

Well, I think there is already an extra dash, the package creator left out the second extra dash, if following the camel case rule. In any event, I used `spack list` to search for packages but did not account for extra '-' characters.

Reply:
For BioPerl, I had a dependency for `perl-uri`. The older package has a dependency for `perl-uri-escape`. However, `URI::Escape` is in the URI package, so in spack, `perl-uri` and `perl-uri-escape` are the same package, but at different versions.

```
class PerlUri(PerlPackage):
    ...
    homepage = "http://search.cpan.org/~ether/URI-1.72/lib/URI.pm"
    url      = "http://search.cpan.org/CPAN/authors/id/E/ET/ETHER/URI-1.72.tar.gz"

    version('1.72', 'cd56d81ed429efaa97e7f3ff08851b48')
```
```
class PerlUriEscape(PerlPackage):
    ...
    homepage = "https://metacpan.org/pod/URI::Escape"
    url      = "https://cpan.metacpan.org/authors/id/E/ET/ETHER/URI-1.71.tar.gz"

    version('1.71', '247c3da29a794f72730e01aa5a715daf')
```
Unless there is a reason not to, I will create a separate PR to remove `perl-uri-escape` and replace it in `depends_on` statements with `perl-uri`.

Reply:
I pushed an update for `perl-bio-perl`. Since the version of 1.007002 was

1. really 1.7.2
1. seen by spack as greater than the current 1.7.6
1. and version constraints needed to be specified

I converted version 1.007002 to 1.7.2. Let me know if that was a bad idea. Also, because the configure method for the older version was overridden I added a `@when` decorator. However, for some reason, with that in place, the `@run_after(configure) decorator in `PerlPackage` was not getting picked up. I opted to override the build and install methods as well for the older version.